src/read/coff/import.rs

```diff
@@ -0,0 +1,208 @@
+//! Support for reading short import files.
+//!
+//! These are used by some Windows linkers as a more compact way to describe
+//! dynamically imported symbols.
+
+use crate::read::{Architecture, Error, ReadError, ReadRef, Result};
+use crate::{pe, ByteString, Bytes, LittleEndian as LE};
+
+/// A Windows short form description of a symbol to import.
+/// Used in Windows import libraries. This is not an object file.
+#[derive(Debug, Clone)]
+pub struct MsImportFile<'data> {
+    header: &'data pe::ImportObjectHeader,
+    dll: ByteString<'data>,
+    symbol: ByteString<'data>,
+    kind: ImportType,
+    import: Option<ByteString<'data>>,
+}
+impl<'data> MsImportFile<'data> {
+    /// Parse it.
+    pub fn parse<R: ReadRef<'data>>(data: R) -> Result<Self> {
+        let mut offset = 0;
+        let header = pe::ImportObjectHeader::parse(data, &mut offset)?;
+
+        let mut strings = Bytes(
+            data.read_bytes(&mut offset, header.data_size() as u64)
+                .read_error("Not enough data")?,
+        );
+        let symbol = strings
+            .read_string()
+            .read_error("Could not read symbol name")?;
+        let dll = strings
+            .read_string()
+            .read_error("Could not read DLL name")?;
+
+        // Unmangles a name by removing a `?`, `@` or `_` prefix.
+        fn strip_prefix(s: &[u8]) -> &[u8] {
+            match s.split_first() {
+                Some((b'?' | b'@' | b'_', rest)) => rest,
+                _ => s,
+            }
+        }
+        Ok(Self {
+            header,
+            dll: ByteString(dll),
+            symbol: ByteString(symbol),
+            kind: header.import_type()?,
+            import: match header.name_type()? {
+                ImportNameType::Ordinal => None,
+                ImportNameType::Name => Some(symbol),
+                ImportNameType::NameNoPrefix => Some(strip_prefix(symbol)),
+                ImportNameType::NameUndecorate => {
+                    Some(strip_prefix(symbol).split(|&b| b == b'@').next().unwrap())
+                }
+                ImportNameType::NameExportAs => Some(
+                    strings
+                        .read_string()
+                        .read_error("Could not read export name")?,
```

Comment:
How sure are you that this third string exists? If we're just guessing about this based on a vague comment, then I prefer not to try parsing this for now.

Reply:
I've just tested it using a similar method as to the above. The third string does indeed exist. I tested `'C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.35.32215\lib\arm64\pgort.lib'`.

### dumpbin

```
  Version      : 0
  Machine      : A641 (ARM64EC)
  TimeDateStamp: F4086DB3
  SizeOfData   : 0000002A
  DLL name     : pgort140.DLL
  Symbol name  : #IrtAutoSweepA
  Type         : code
  Name type    : exportas
  Hint         : 0
  Name         : IrtAutoSweepA
```

### objdump

```
Format: Short Import File
Architecture: Unknown
DLL: "pgort140.DLL"
Symbol: "#IrtAutoSweepA"
Import: Name("IrtAutoSweepA")
Type: Code
```

Reply:
Thanks. I take it the reason they used `exportas` here is because the leading `#` needs to be stripped, and the other options don't do that.